tests/test_client.py

```diff
@@ -678,6 +678,36 @@ def test_host_key_negotiation_3(self):
     def test_host_key_negotiation_4(self):
         self._client_host_key_good(paramiko.RSAKey, "rsa.key")
 
+    def test_server_hostkey_name(self):
+        hostname = "a-different.name"
+
+        def _test(**connect_args):
+            threading.Thread(target=self._run).start()
+
+            self.tc = SSHClient()
+            self.tc.set_missing_host_key_policy(paramiko.RejectPolicy())
+            kfile = "ecdsa-256.key"
+            host_key = paramiko.ECDSAKey.from_private_key_file(_support(kfile))
+            known_hosts = self.tc.get_host_keys()
+            known_hosts.add(hostname, host_key.get_name(), host_key)
+
+            self.tc.connect(password="pygmalion", **connect_args)
+
+        # make sure there is no hostname for the default
+        self.assertRaises(
+            paramiko.ssh_exception.SSHException,
```

Comment:
Is there any (good) way to narrow the scope of this catch?

`SSHException` is a pretty large umbrella at the moment.

Reply:
I don't think so, but I don't know the code-base well enough to be sure.
This is the exception coming out of `_test(**self.connect_kwargs)`:
```
paramiko.ssh_exception.SSHException: Server '[127.0.0.1]:60469' not found in known_hosts
```

Reply:
That's pretty much what I figured. This should be fine, especially in a test.